src/main/java/rx/Single.java

```diff
@@ -1997,4 +2025,46 @@ public void call(SingleSubscriber<? super T> singleSubscriber) {
             }
         });
     }
+
+    /**
+     * FOR INTERNAL USE ONLY.
+     * <p>
+     * Converts {@link Iterable} of {@link Single} to array of {@link Single}.
+     *
+     * @param singlesIterable
+     *         non null iterable of {@link Single}.
+     * @return array of {@link Single} with same length as passed iterable.
+     */
+    @SuppressWarnings("unchecked")
+    static <T> Single<? extends T>[] iterableToArray(final Iterable<? extends Single<? extends T>> singlesIterable) {
+        final Single<? extends T>[] singlesArray;
+        int count;
+
+        if (singlesIterable instanceof List) {
+            List<? extends Single<? extends T>> list = (List<? extends Single<? extends T>>) singlesIterable;
+            count = list.size();
+            singlesArray = list.toArray(new Single[count]);
+        } else {
+            Single<? extends T>[] tempArray = new Single[8]; // Magic number used just to reduce number of allocations.
+            count = 0;
+            for (Single<? extends T> s : singlesIterable) {
+                if (count == tempArray.length) {
+                    Single<? extends T>[] sb = new Single[count + (count >> 2)];
+                    System.arraycopy(tempArray, 0, sb, 0, count);
+                    tempArray = sb;
+                }
+                tempArray[count] = s;
+                count++;
+            }
+
+            if (tempArray.length == count) {
+                singlesArray = tempArray;
+            } else {
+                singlesArray = new Single[count];
+                System.arraycopy(tempArray, 0, singlesArray, 0, count);
```

Comment:
Oh, got it, so it was my bug when I split your code into two parts, sorry. 

I can return `Map.Entry` from `iterableToArray` where one value will be the array and another — real `count`, but personally I'd prefer current variant.

If you afraid about performance — we can add more general check `instanceof Collection` and initialize the array with correct size even if `Iterable` is not a `List`. For real, I don't even remember when I used something that does not implement `Collection` but implements `Iterable`, I guess will cover a lot of cases with such check: `List`, `Set`, `Queue` and so on.